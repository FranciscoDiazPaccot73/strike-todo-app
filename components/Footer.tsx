import Image from "next/image";

import { FOOTER } from "./utils/constants";

const Footer = () => (
  <footer className="pt-5 absolute bottom-0 flex items-center justify-center w-full border-t border-gray-40">
    <div className='flex items-center flex-1 justify-center pt-4 pb-10 relative text-white'>
      <div className="text-sm flex flex-col items-center justify-center" >
        <a className="flex items-center" href="https://strike.sh/en" target="_blank" rel="noopener noreferrer">
          <span className="mr-2">{FOOTER.CHALLENGE}</span>
          <Image
            src="/strike-logo.webp"
            alt="Strike webpage"
            width={84}
            height={32}
          />
        </a>
      </div>
      <div className='absolute bottom-3 text-[10px] right-8 flex'>
        {FOOTER.CREATED_BY} <a className="font-bold ml-1" href='https://franciscodiazpaccot.dev' target="_blank" rel="noreferrer noopener">
        {FOOTER.AUTHOR}</a>
      </div>
    </div>
  </footer>
)

export default Footer;
