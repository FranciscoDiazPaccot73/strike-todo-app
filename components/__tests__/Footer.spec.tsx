import React from 'react';
import { render } from '@testing-library/react'
import Footer from "../Footer";

describe("Footer", () => {
  it("Should render Footer component", () => {
    const container = render(<Footer />);
    expect(container).toMatchSnapshot()
  });
});
