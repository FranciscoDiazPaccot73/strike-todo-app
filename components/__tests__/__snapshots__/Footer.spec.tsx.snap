// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should render Footer component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="pt-5 absolute bottom-0 flex items-center justify-center w-full border-t border-gray-40"
      >
        <div
          class="flex items-center flex-1 justify-center pt-4 pb-10 relative text-black dark:text-white"
        >
          <div
            class="text-sm flex flex-col items-center justify-center"
          >
            <a
              class="flex items-center"
              href="https://strike.sh/en"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="mr-2"
              >
                Frontend Technical challenge for
              </span>
              <img
                alt="Strike webpage"
                data-nimg="1"
                decoding="async"
                height="32"
                loading="lazy"
                src="/_next/image?url=%2Fstrike-logo.webp&w=256&q=75"
                srcset="/_next/image?url=%2Fstrike-logo.webp&w=96&q=75 1x, /_next/image?url=%2Fstrike-logo.webp&w=256&q=75 2x"
                style="color: transparent;"
                width="84"
              />
            </a>
          </div>
          <div
            class="absolute bottom-3 text-[10px] right-8 flex"
          >
            Created By
             
            <a
              class="font-bold ml-1"
              href="https://franciscodiazpaccot.dev"
              rel="noreferrer noopener"
              target="_blank"
            >
              Francisco Diaz Paccot
            </a>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="pt-5 absolute bottom-0 flex items-center justify-center w-full border-t border-gray-40"
    >
      <div
        class="flex items-center flex-1 justify-center pt-4 pb-10 relative text-black dark:text-white"
      >
        <div
          class="text-sm flex flex-col items-center justify-center"
        >
          <a
            class="flex items-center"
            href="https://strike.sh/en"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="mr-2"
            >
              Frontend Technical challenge for
            </span>
            <img
              alt="Strike webpage"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fstrike-logo.webp&w=256&q=75"
              srcset="/_next/image?url=%2Fstrike-logo.webp&w=96&q=75 1x, /_next/image?url=%2Fstrike-logo.webp&w=256&q=75 2x"
              style="color: transparent;"
              width="84"
            />
          </a>
        </div>
        <div
          class="absolute bottom-3 text-[10px] right-8 flex"
        >
          Created By
           
          <a
            class="font-bold ml-1"
            href="https://franciscodiazpaccot.dev"
            rel="noreferrer noopener"
            target="_blank"
          >
            Francisco Diaz Paccot
          </a>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
